src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -120,54 +151,35 @@ public boolean validCheckForObject(final AtlasObject object)
         if (direction.equals(RoundaboutDirection.CLOCKWISE) && !isLeftDriving
                 || direction.equals(RoundaboutDirection.COUNTERCLOCKWISE) && isLeftDriving)
         {
-            return Optional.of(this.createFlag(new HashSet<>(roundaboutEdges),
-                    this.getLocalizedInstruction(0, edge.getOsmIdentifier())));
+            instructions.add(this.getLocalizedInstruction(0));
+        }
+
+        // If there are car navigable edges inside the roundabout flag it, as it is probably
+        // malformed or a throughabout
+        if (roundaboutEdgeSet.stream().noneMatch(this::ignoreBridgeTunnelCrossings)
+                && this.roundaboutEnclosesRoads(roundaboutEdges))
+        {
+            instructions.add(this.getLocalizedInstruction(2));
+        }
+
+        if (!instructions.isEmpty())
+        {
+            return Optional.of(this.createFlag(roundaboutEdgeSet, String.join(" ", instructions)));
```

Comment:
can you add each instruction individually so that they show up in separate lines in Maproulette.